include/deal.II/fe/fe_values.h

```diff
@@ -1663,9 +1663,9 @@ class FEValuesBase : protected FEValuesData<dim,spacedim>,
    *
    * @dealiiRequiresUpdateFlags{update_values}
    */
-  template <class InputVector, typename number>
+  template <class InputVector>
   void get_function_values (const InputVector &fe_function,
-                            std::vector<number> &values) const;
+                            std::vector<typename InputVector::value_type> &values) const;
```

Comment:
That's a fair point. On the other hand, I think what happens equally often is that people call these functions with different template types but there is no instantiation -- in which case you get an obscure linker error rather than a compiler error.